File: src/main/java/fr/umontpellier/iut/dominion/cards/component/CardSelector.java
package fr.umontpellier.iut.dominion.cards.component;

import fr.umontpellier.iut.dominion.Player.Player;
import fr.umontpellier.iut.dominion.cards.Card;

import java.util.List;

/**
 *Interface fonctionnelle de selection d'une carte
 */
@FunctionalInterface
public interface CardSelector {
    Card select(Player attacker, Player victim, List<Card> options);
}

File: src/main/java/fr/umontpellier/iut/dominion/cards/component/CardSelector.kt
package fr.umontpellier.iut.dominion.cards.component

import fr.umontpellier.iut.dominion.Player.Player
import fr.umontpellier.iut.dominion.cards.Card

/**
 * Interface fonctionnelle de selection d'une carte
 */
fun interface CardSelector : suspend (Player, Player, MutableList<Card>) -> Card? {
    override suspend fun invoke(chooser: Player, victim: Player, options: MutableList<Card>): Card?
}
